lib/server.js

```diff
@@ -164,6 +184,64 @@ module.exports = function createServer(config, log) {
   )
 
   api.post(
+    '/checkAuthenticated',
+    function (req, res, next) {
+      var action = req.body.action
+      var email = req.body.email
+      var ip = req.body.ip
+      var uid = req.body.uid
+
+      if(!action || !ip || !uid){
+        var err = {code: 'MissingParameters', message: 'action, email, ip and uid are all required'}
```

Comment:
just to be clear: not worry about not rate limiting allowedEmailDomains, right? that is, let restmail be rate limited the same as all the others.